Update and improve README.md (#403)
willowtreeapps/assertk: README.md

> # AssertK

Author: Good to know! I was wondering what your thoughts were on that. I'll update it elsewhere in the PR.

[![CircleCI](https://circleci.com/gh/willowtreeapps/assertk.svg?style=svg)](https://circleci.com/gh/willowtreeapps/assertk)
[![Maven Central](https://img.shields.io/maven-central/v/com.willowtreeapps.assertk/assertk.svg)](https://search.maven.org/search?q=g:com.willowtreeapps.assertk)
[![Sonatype Snapshot](https://img.shields.io/nexus/s/https/oss.sonatype.org/com.willowtreeapps.assertk/assertk.svg)](https://oss.sonatype.org/content/repositories/snapshots/com/willowtreeapps/assertk/)

AssertK is a fluent assertion library for Kotlin inspired by [AssertJ](https://github.com/assertj/assertj-core).

* [Documentation](https://willowtreeapps.github.io/assertk/assertk/assertk.assertions/index.html)

## Why another assertion library?

> You might be asking, "If AssertJ already exists, why create another library?". It's true, AssertK is very similar to AssertJ. But AssertK is written in Kotlin. So it has one major advantage: extension methods.
> 
> Every assertion in AssertK is simply an extension method. This means adding your own **custom assertions** is easy. Sure, you _can_ create custom assertions in AssertJ, but they're cumbersome and not very intuitive.
> 
> See [Custom Assertions](#custom-assertions) below to find out how to do this.

Reviewer: 👍 I do like this copy. The one thing I would change is calling assertj "cumbersome and not very intuitive", would rather focus on bringing up the benefits than highlighting negatives. Maybe something like

```suggestion
You might be asking, "If AssertJ already exists, why create another library?". It's true, AssertK is very similar to AssertJ. But AssertK is written in Kotlin. So it has one major advantage: extension methods. This makes adding your own assertion methods far simpler.

See [Custom Assertions](#custom-assertions) below to find out how to do this.
```
